### PR TITLE
Working post

### PR DESCRIPTION
**Pull request by mirosval**

### CODE REVIEW

File: server/Cargo.toml [hunk 1]
```diff
@@ -8,9 +8,11 @@ edition = "2018"
 
 [dependencies]
 chrono = { version = "0.4", features = ["serde"] }
-diesel = { version = "1.0.0", features = ["postgres", "numeric", "chrono"] }
+diesel = { version = "1.0.0", features = ["postgres", "numeric", "chrono", "r2d2"] }
```
Comment (ChetanBhasin) [inline server/Cargo.toml]: What's r2d2?

File: server/migrations/2020-03-20-224603_create_checkin_table/up.sql [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 CREATE TABLE IF NOT EXISTS checkins (
     id SERIAL PRIMARY KEY,
-    gps_lat POINT NOT NULL, 
+    gps POINT NOT NULL,
```
Comment (ChetanBhasin) [inline server/migrations/2020-03-20-224603_create_checkin_table/up.sql]: Could you try this after enabling the postGIS extension too? If not, I'll do it later.

File: server/src/main.rs [hunk 3]
```diff
@@ -1,61 +1,69 @@
...
-pub async fn establish_connection() -> Db {
+fn get_connection_pool() -> Pool {
     dotenv().ok();
-
-    let database_url = env::var("DATABASE_URL")
-        .expect("DATABASE_URL must be set");
-    let pgc = PgConnection::establish(&database_url)
-        .expect(&format!("Error connecting to {}", database_url));
-    Arc::new(Mutex::new(pgc))
+    let database_url = env::var("DATABASE_URL").expect("DATABASE_URL must be set");
```
Comment (ChetanBhasin) [inline server/src/main.rs]: I don't want to use `.expect` really. Do you think it'd be a lot of work to introduce a service error?

File: server/Cargo.toml (hunk 1 above)
Comment (mirosval) [reply server/Cargo.toml]: connection pool

File: server/src/main.rs (hunk 3 above)
Comment (mirosval) [reply server/src/main.rs]: I mean this should just prevent the service from starting no?

**Pull request merged**